src/main/java/com/example/solidconnection/mentor/service/MentoringCommandService.java

```diff
@@ -0,0 +1,103 @@
+    private final MentorRepository mentorRepository;
+
+    @Transactional
+    public MentoringApplyResponse applyMentoring(long siteUserId, MentoringApplyRequest mentoringApplyRequest) {
+        validateAlreadyMentor(siteUserId);
+
+        Mentoring mentoring = Mentoring.builder()
+                .mentorId(mentoringApplyRequest.mentorId())
+                .menteeId(siteUserId)
+                .verifyStatus(VerifyStatus.PENDING)
+                .build();
+
+        return MentoringApplyResponse.from(mentoringRepository.save(mentoring));
+    }
+
+    private void validateAlreadyMentor(long siteUserId) {
+        if (mentorRepository.existsBySiteUserId(siteUserId)) {
+            throw new CustomException(ALREADY_MENTOR);
+        }
+    }
+
+    @Transactional
+    public MentoringConfirmResponse confirmMentoring(long siteUserId, long mentoringId, MentoringConfirmRequest mentoringConfirmRequest) {
+        Mentoring mentoring = mentoringRepository.findById(mentoringId)
+                .orElseThrow(() -> new CustomException(MENTORING_NOT_FOUND));
+
+        Mentor mentor = mentorRepository.findById(mentoring.getMentorId())
+                .orElseThrow(() -> new CustomException(MENTOR_NOT_FOUND));
+
+        validateUnauthorizedMentoring(siteUserId, mentor);
+        validateAlreadyConfirmed(mentoring);
+
+        if (mentoringConfirmRequest.status() == VerifyStatus.REJECTED
+                && (mentoringConfirmRequest.rejectedReason() == null || mentoringConfirmRequest.rejectedReason().isBlank())) {
+            throw new CustomException(REJECTED_REASON_REQUIRED);
+        }
+
+        mentoring.confirm(mentoringConfirmRequest.status(), mentoringConfirmRequest.rejectedReason());
+
+        if (mentoringConfirmRequest.status() == VerifyStatus.APPROVED) {
+            mentor.increaseMenteeCount();
+        }
+
+        return MentoringConfirmResponse.from(mentoring);
+    }
+
+    private void validateAlreadyConfirmed(Mentoring mentoring) {
+        if (mentoring.getVerifyStatus() != VerifyStatus.PENDING) {
+            throw new CustomException(MENTORING_ALREADY_CONFIRMED);
+        }
+    }
+
+    @Transactional
+    public MentoringCheckResponse checkMentoring(long siteUserId, long mentoringId) {
+        Mentoring mentoring = mentoringRepository.findById(mentoringId)
+                .orElseThrow(() -> new CustomException(MENTORING_NOT_FOUND));
+
+        Mentor mentor = mentorRepository.findById(mentoring.getMentorId())
+                .orElseThrow(() -> new CustomException(MENTOR_NOT_FOUND));
+
+        validateUnauthorizedMentoring(siteUserId, mentor);
+
+        mentoring.check();
+
+        return MentoringCheckResponse.from(mentoring.getId());
+    }
+
+    // 멘토는 본인의 멘토링에 대해 confirm 및 check해야 한다.
+    private void validateUnauthorizedMentoring(long siteUserId, Mentor mentor) {
+        if (siteUserId != mentor.getSiteUserId()) {
+            throw new CustomException(UNAUTHORIZED_MENTORING);
+        }
+    }
```

Comment:
주석으로 적혀있는 "멘토는 본인의 멘토링에 대해 confirm 및 check해야 한다."의 내용과 함수의 내용이 호응되지 않는 것 같아요..!

주석에 맞는 구현을 하려면 이런식이어야 할 것 같습니다.
```suggestion
    private void validateUnauthorizedMentoring(Mentor mentor, Mentoring mentoring) {
        if (mentoring.getMentorId() != mentor.getId()) {
            throw new CustomException(UNAUTHORIZED_MENTORING);
        }
    }
```

제가 잘못 이해하고 있는거라면 말씀해주세요!

Reply:
주석 내용과 메서드 동작이 매칭되게끔 변경하였습니다 !
저도 주석 보고 메서드보니 한참 고민했네요 ..